internal/vulnerability/snapshot.go

```diff
@@ -109,3 +144,33 @@ func runWithGrace(ctx context.Context, grace time.Duration, f func(ctx context.C
 	defer stop() // if the callback finishes in time, stop the AfterFunc
 	f(newCtx)    // finally, call the actual callback!
 }
+
+type contextualChan[T any] struct {
+	ch chan T
+}
```

Comment:
[ just sharing thoughts - not suggesting to change it ]
I wonder which one of "wrapper struct vs just generic functions", is leaner in that case. 

eg instead of a wrapper just having something like that
```golang
func SendToChannel[T any](ctx context.Context, ch chan<- T, t T) bool
func ReadFromChannel[T any](ctx context.Context, ch <-chan T) (T, bool)
```

Reply:
I wanted to completely avoid exposing the channel to prevent misusing it. Both can be valid approaches if direct access to the channel is needed.